### PR TITLE
[modified] prevent placement of seeds on non-growing tiles. Seeds fall down when losing supporting tile

### PR DESCRIPTION
**Pull request by Nananas**
## Status
- **READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Placement of seeds now only allowed on tiles where seeds can actually grow.
Seeds should have improved physics, which means they will fall down when their support is lost.
This should also fix issue #606.

## Steps to Test or Reproduce

Try and fail planting seeds on stone, wood, platforms, shop etc.
Plant a seed on ground tile (only place you can put them now), then dig out that tile and the seed will drop down (with a slight delay).

### CODE REVIEW

File: Entities/Common/Building/PlacementCommon.as [hunk 1]
```diff
@@ -182,6 +183,14 @@ bool isBuildableAtPos(CBlob@ this, Vec2f p, TileType buildTile, CBlob @blob, boo
 				}
 			}
 		}
+
+
+		if (isSeed)
+		{
+			// from canGrow.as
+			return (map.isTileGroundStuff(map.getTile(p + Vec2f(0, 8)).type));
```
Comment (Harrison-Miller) [inline Entities/Common/Building/PlacementCommon.as:191]: This includes dirt back wall. I would narrow it down so you can only place it on dirt.

File: Entities/Natural/Seed/Seed.as [hunk 2]
```diff
@@ -136,4 +136,25 @@ void onTick(CBlob@ this)
 			}*/
 		}
 	}
+
+	if (isServer())
```
Comment (Harrison-Miller) [inline Entities/Natural/Seed/Seed.as:140]: checkout FallOnNoSupport.as. You might want to look into using that script. Or using onBlobCollapse hook.

File: Entities/Natural/Seed/Seed.as [hunk 3]
```diff
@@ -136,4 +136,25 @@ void onTick(CBlob@ this)
...
+			// drop down when not supported
+			CShape@ shape = this.getShape();
+			shape.server_SetActive(true);
+
+			shape.SetStatic(false);
+		}
+	}
+}
+
+bool doesCollideWithBlob(CBlob@ this, CBlob@ blob)
```
Comment (Harrison-Miller) [inline Entities/Natural/Seed/Seed.as:157]: Good catch.

File: Entities/Common/Building/PlacementCommon.as (hunk 1 above)
Comment (Nananas) [reply Entities/Common/Building/PlacementCommon.as]: Should be fixed in latest commit

**Comment by Harrison-Miller** [comment]
gonna call this good, probably don't need to worry about how the falling logic is done.

**Pull request merged**